kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/HttpClientUtils.java

```diff
@@ -139,6 +139,16 @@ public Response intercept(Chain chain) throws IOException {
                 }
             }
 
+            if (config.getUserAgent() != null && !config.getUserAgent().isEmpty()) {
+                httpClient.interceptors().add(new Interceptor() {
```

Comment:
I'm not familiar with the difference, this is the only documentation on okHttp interceptors I can find: http://square.github.io/okhttp/3.x/okhttp/